### ISSUE TITLE
How does MegaQC determine the date (x axis) for each report uploaded?

### ISSUE THREAD

**Issue by chrispyatt** [ewels/MegaQC#422]
Our previous installation of MegaQC seemed to mostly work as expected, with each tranche of data appearing on the x axis approximately on the date of sequencing/analysis (so we could see what changed between runs). However, having now restored from backup on a new system, all data-points are being plotted on the same date (corresponding to the config_creation_date field in multiqc_data.json). 

I don't believe I've changed any settings so what is the mechanism for deciding where on the x axis data should sit? I don't see any other dates within the JSON, and the upload dates are still the same as before (as the DB was restored rather than recreated), so I'm a bit confused about what could have changed.

I also note that recreating the database on a separate system does create correct plots (with plausible x axes) so I don't think the JSONs are wrong. I'd like to avoid doing this on the production version as the original database has a lot of saved filters that I'd have to remake manually.

**Comment by multimeric** [comment]
At the point where MegaQC ingests the reports, it chooses either the `config_creation_date`, or a manually provided date if one was passed in via the CLI or REST API. Then this date gets saved to the DB. 

So I'm not sure why this particular issue happened to you. How did you restore the database? Also, can you run an SQL query to look at the data? Something like `SELECT report_hash, created_at FROM report LIMIT 50;` or apply some kind of date filter like `WHERE created_at BETWEEN DATE '2020-02-16' AND DATE '2021-02-16'` if you know the time period affected.

**Comment by multimeric** [comment]
> Perhaps the long term fix is to force it and supply dates along with the upload then

Yes, I would recommend this if you have more precise metadata available to you

> I don't understand why I'm seeing different behaviour vs previous deployment.

I don't either, but I think you're most likely right that not all data is being loaded correctly. If you investigate further please let me know what you find.
